### PR TITLE
enable CPU map and multiple streams for ARM Linux

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *enable CPU map for ARM Linux based on frequency information*
 - *enable multiple streams for ARM Linux*
 - *TODO CVS-112149*

### Tickets:
 - *CVS-112143*

### CODE REVIEW

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 1]
```diff
@@ -231,4 +295,95 @@ void parse_processor_info_linux(const int _processors,
...
+                UpdateProcMapping(core_1);
+
+                _cpu_mapping_table[core_1][CPU_MAP_GROUP_ID] = _cores;
+            }
+
+            _cores++;
+        }
+    }
+
+    if (_sockets == 1) {
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp]: updated

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 2]
```diff
@@ -231,4 +297,95 @@ void parse_processor_info_linux(const int _processors,
...
+                core_2 = std::stoi(sub_str);
+
+                UpdateProcMapping(core_1);
+
+                _cpu_mapping_table[core_2][CPU_MAP_PROCESSOR_ID] = core_2;
+                _cpu_mapping_table[core_2][CPU_MAP_SOCKET_ID] = _cpu_mapping_table[core_1][CPU_MAP_SOCKET_ID];
+                _cpu_mapping_table[core_2][CPU_MAP_CORE_ID] = _cpu_mapping_table[core_1][CPU_MAP_CORE_ID];
+                _cpu_mapping_table[core_2][CPU_MAP_CORE_TYPE] = HYPER_THREADING_PROC;
+                _cpu_mapping_table[core_2][CPU_MAP_GROUP_ID] = _cpu_mapping_table[core_1][CPU_MAP_GROUP_ID];
+            } else if (pos != std::string::npos) {
```
Comment (xufang-lisa) [inline src/inference/src/os/lin/lin_system_conf.cpp:359]: here the pos should be the initial value 0?

File: src/inference/tests/unit/cpu_map_parser.cpp [hunk 3]
```diff
@@ -572,18 +574,112 @@ LinuxCpuMapTestCase _1sockets_6cores_hyperthreading = {
...
+        {"2", "0", "1800000"},
+        {"3", "0", "1800000"},
+    },
+};
+
+TEST_P(LinuxCpuMapFreqParserTests, LinuxCpuMapFreq) {}
+
+INSTANTIATE_TEST_SUITE_P(CPUMap,
+                         LinuxCpuMapFreqParserTests,
+                         testing::Values(freq_2sockets_20cores_hyperthreading, freq_1sockets_4cores));
```
Comment (sunxiaoxia2022) [inline src/inference/tests/unit/cpu_map_parser.cpp:681]: I think it is lack of "Pcore + Ecore" platform test case.

**Review by riverlijunjie — APPROVED** [review]
LGTM

**Review by xufang-lisa — APPROVED** [review]

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 4]
```diff
@@ -231,4 +297,91 @@ void parse_processor_info_linux(const int _processors,
...
+                           int& _sockets,
+                           int& _cores,
+                           std::vector<std::vector<int>>& _proc_type_table,
+                           std::vector<std::vector<int>>& _cpu_mapping_table) {
+    std::vector<int> freq_list = {0, 0, 0, 0};
+
+    _processors = system_info_table.size();
+    _cpu_mapping_table.resize(_processors, std::vector<int>(CPU_MAP_TABLE_SIZE, -1));
+
+    auto UpdateProcMapping = [&](const int nproc) {
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp]: updated

File: src/inference/tests/unit/cpu_map_parser.cpp (hunk 3 above)
Comment (wangleis) [reply src/inference/tests/unit/cpu_map_parser.cpp]: updated

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 5]
```diff
@@ -22,35 +22,103 @@ CPU::CPU() {
     std::vector<std::vector<std::string>> system_info_table;
 
     _num_threads = parallel_get_max_threads();
-    auto GetCatchInfoLinux = [&]() {
-        _processors = sysconf(_SC_NPROCESSORS_ONLN);
-        system_info_table.resize(_processors, std::vector<std::string>(3));
+    auto GetCacheInfoLinux = [&]() {
```
Comment (sunxiaoxia2022) [inline src/inference/src/os/lin/lin_system_conf.cpp:25]: I think this name and "GetFreqInfoLinux" should be changed by under score case.

Comment (wangleis) [reply src/inference/src/os/lin/lin_system_conf.cpp:25]: updated

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 6]
```diff
@@ -292,6 +292,11 @@ void Engine::GetPerformanceStreams(Config& config, const std::shared_ptr<ngraph:
     const auto latency_name = std::string(CONFIG_VALUE(LATENCY)) + "_" + std::string(ov::num_streams.name());
     const auto tput_name = std::string(CONFIG_VALUE(THROUGHPUT)) + "_" + std::string(ov::num_streams.name());
 
+#if defined(OPENVINO_ARCH_ARM) || defined(OPENVINO_ARCH_ARM64)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/plugin.cpp:295]: We already force num_streams to be equal 1 in CPU config. Why it should be updated here?

Comment (wangleis) [reply src/plugins/intel_cpu/src/plugin.cpp:295]: This WA for CI checks will be removed after export/import refactor for MT 2.0 which will be track in CVS-112149

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/plugin.cpp:295]: Please put this sentence as comment in the code

Comment (wangleis) [reply src/plugins/intel_cpu/src/plugin.cpp:295]: updated

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**